indico/modules/events/abstracts/models/comments.py

```diff
@@ -17,9 +17,19 @@
 from __future__ import unicode_literals
 
 from indico.core.db import db
-from indico.core.db.sqlalchemy import UTCDateTime
+from indico.core.db.sqlalchemy import PyIntEnum, UTCDateTime
 from indico.util.date_time import now_utc
 from indico.util.string import format_repr, return_ascii, text_to_repr
+from indico.util.struct.enum import IndicoEnum
+
+
+class AbstractCommentVisibility(int, IndicoEnum):
```

Comment:
As discussed with @pferreir, we want at least the 3 first ones, with potential use for the fourth. At this point I think there's no harm for having the totally public visibility as well.